tasks/async/async.js

```diff
@@ -9,36 +9,57 @@ const {
  * console.log data got from callbackFunction
  */
 function printDataCallback() {
+    function printText(str, data) {
+        console.log(data);
+    }
+    callbackFunction(printTesxt);
 }
 
 /**
  * console.log data got from promiseFunction
  */
 function printDataPromise() {
+    promiseFunction()
+        .then(data => {
+            console.log(data);
+        });
 }
 
 /**
  * console.log data got from promiseFunction using async/await
  */
 async function printDataAsyncAwait() {
+const data = await promiseFunction();
+  console.log(data);
 }
 
 /**
  * throw error data from callbackFunctionError
  */
 function handleErrorCallback() {
-}
+    function throwErr(err) {
+    throw new Error ('oops');
+  }
+  callbackFunctionError(throwErr);
+  }
 
 /**
  * throw error come from promiseFunctionError
  */
 function handlePromiseError() {
-}
-
+    try{
```

Comment:
it is a solution for the next function handleAsyncAwaitError.
here you should catch the promise's error:
promiseFunctionError.catch(err => console.log(err));